packages/api/src/hooks/useAuthorize.ts

```diff
@@ -11,10 +11,9 @@ const useAuthorize = () => {
     const current_token = getActiveAuthTokenIDFromLocalStorage();
     const invalidate = useInvalidateQuery();
     const { switchEnvironment } = useAPIContext();
-
     const { data, ...rest } = useQuery('authorize', {
         payload: { authorize: current_token || '' },
-        options: { enabled: Boolean(current_token) },
+        options: { enabled: Boolean(current_token), staleTime: 1000 * 60 },
```

Comment:
also - this call have side effects on the server, 
as a rule of thumb, calls with side effects on server, shouldn't be cached on frontend level

Reply:
We need to cache it because we need the response from this call. And we dont need to authorize everytime we need the data right, unless it changed. We will re-fetch/invalidate it if it changes

Reply:
do you need response, or do you need the websocket to switch context to new account?
what if underlying websocket got reconnected but you requested authorise, but it got cached on frontend level?
I can imagine other scenarios when caching authorise request on frontend level might mess up with the server-side-authorise and context swtiching.

Reply:
reason I'm commenting out, is that we already have race hazards and bug resulting from the fact that we control requests and data flow poorly and that its quite unpredictable, so adding extra layer of complexity sounds won't make debugging and maintaining easier. 

I would be much more keen to uncover the reasons for authorise call being called multiple times than just cache it and hide the actuall underlying problem

Reply:
for calls which does have side effects on the server level, like authorise, we shouldnt' cache on frontend, 
for calls which do not have side effects on server (e.g. fetching data) we can cache them, as long as we're happy with data being stale (e.g. imagine that you added account, got new accounts list with new account, but balance/currency info and other requests are still cached and thus do not provide data for newly added account intermittenly - so does not work only when you previously have requested them in last 60s)

Reply:
so ultimately - cache on can be applied only for requests which are stale and we're sure that its fine for the app to receive potentially stale data, and thats to make sure we wont have race hazards/intermittent bugs.

Reply:
it does make sense to switch to different endpoint in new library, thats how it should be, 
but in this verions - we do use the same endpoint for both, so we cannot just cache it, too many edge cases and problems (e.g. when adding accounts)

I do not feel like chasing elusive bugs later on, like the ones we have now - e.g. occasional whitescreen, occasional application being stuck on whitescreen, occasionaly displaying incorrect currency - thats all resulting from poor data flow / data handling.